app/src/components/Blog.jsx

```diff
@@ -1,17 +1,36 @@
 import React, { Component } from 'react';
 import Footer from './Shared/Footer';
 import { Link } from 'react-router';
-import listposts from '../../styles/components/c-list-posts.scss';
+import listPosts from '../../styles/components/c-list-posts.scss';
 import CoverPrimary from './Shared/CoverPrimary';
 import PaginationBlog from './Blog/Pagination';
 import boxtriangle from '../../assets/icons/box_triangle.svg';
 
 class Blog extends Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      totalPageCount: 0
+    };
+
+    this.boundOnPageChange = this.onPageChange.bind(this);
```

Comment:
maybe it's a im-a-noob-js-dev-so-this-is-not-really-a-smell kind of smell, you know. It's not about how it's used, it's about code readability. The JS interpreter overloads the unbound method with the bound one, just that a human might confuse one for another when reading the code.